### PR TITLE
Removed legacy hudson update site logic

### PR DESCRIPTION
**Pull request by StefanSpieker**
Removed legacy hudson update site logic. The sites do not exist anymore as far as I can tell. Hudson is also archived since  February 2021

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/model/UpdateSite.java
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/UpdateSite.java:534]: This method is overridden in https://github.com/jenkinsci/simpleupdatesite-plugin , and hence it will lead to code compatibility issues. Not sure anyone cares

**Review by oleg-nenashev — COMMENTED** [review]
No objections per se, but just FYI it breaks the public API

File: core/src/main/java/hudson/model/UpdateSite.java
Comment (basil) [reply core/src/main/java/hudson/model/UpdateSite.java:534]: I'm OK with dropping support for this plugin as long as a deprecation notice is added to the Update Center and an entry is added to the changelog/upgrade guide.

Comment (daniel-beck) [reply core/src/main/java/hudson/model/UpdateSite.java:534]: > hence it will lead to code compatibility issues

Why does this break?

Comment (oleg-nenashev) [reply core/src/main/java/hudson/model/UpdateSite.java:534]: The `@Override` definition in simpleupdatesite-plugin will prevent the code from compiling against new Jenkins core versions. There should be no other consequences

**Review by res0nance — APPROVED** [review]

**Comment by StefanSpieker** [comment]
Thanks for your feedback, I think for the moment removing the functionality but keeping it for compatibility is a way to go now. Later we can still remove it completly. What do you think?

**Review by basil — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
